opm/input/eclipse/Schedule/ScheduleState.cpp

```diff
@@ -196,6 +196,67 @@ bool ScheduleState::first_in_year() const {
     return this->m_first_in_year;
 }
 
+bool ScheduleState::well_group_contains_lgr(const Group& grp, const std::string& lgr_tag) const
+{
+    if (grp.wellgroup()){
+        for (const auto& well_name : grp.wells()) {
+            const auto& well = this->wells.get(well_name);
+            if (well.get_lgr_well_tag().value_or("") == lgr_tag) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+bool ScheduleState::group_contains_lgr(const Group& grp, const std::string& lgr_tag) const
+{
+    bool contain_lgr = false;
+    if (grp.wellgroup()){
+        contain_lgr = this->well_group_contains_lgr(grp, lgr_tag);
+    }
+    else {
+        for (const auto& child_group_name : grp.groups()) {
+            const auto& child_group = this->groups.get(child_group_name);
+            if (child_group.wellgroup()){
+                contain_lgr = this->well_group_contains_lgr(grp, lgr_tag);
+            }
+            else {
+                contain_lgr = this->group_contains_lgr(child_group, lgr_tag);
+            }
+        }
```

Comment:
Depending on the intended semantics of `group_contins_lgr()`, this loop may produce unexpected results.  Suppose `grp.groups()` is `{"A", "B", "C"}`, i.e., that `grp` is the parent of three subordinate groups.  Suppose further that `"B"` is a well group and that the local grid represented by `lgr_tag` is below `"B"`.  When processing `"B"`, then `contain_lgr` will be `true`&ndash;at least if we pass `child_group` as an argument to `well_group_contains_lgr()` instead of `grp`&ndash;but `contain_lgr` will be reset to `false` when we then go on to process child group `"C"`.  As a result, `group_contains_lgr()` will return `false` in this case.  Is that what you want?